### PR TITLE
Update readme with best practices & 'Join the convo' section

### PR DESCRIPTION
**Pull request by kelsk**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,16 @@
...
+- YAML authoring support
+- Using a development cluster for local testing
+- Friendly logging tools
+- Integrated debugging
+- Automated local dev loop
+- Custom Samples to share team best practices
+
+## Join the conversation
+
+Interested in Cloud Code? You can try pre-release features, provide valuable feedback, and ask questions by [installing pre-release builds](https://cloud.google.com/code/docs/intellij/insiders#get) and joining the [Cloud Code Insiders group](https://groups.google.com/g/cloud-code-insiders).
```
Comment (abbycar) [inline README.md:14]: I'd put...

Interested in Cloud Code? You can try pre-release features, provide valuable feedback, and ask questions by installing pre-release builds ([IntelliJ](https://cloud.google.com/code/docs/intellij/insiders#get), [VS Code](https://cloud.google.com/code/docs/vscode/insiders#get))....

**Review by abbycar — COMMENTED** [review]
Couple comments, but otherwise LGTM! :D

File: README.md [hunk 2]
```diff
@@ -1,2 +1,16 @@
-# k8s-next-21
-A demo repo for Google Cloud Next '21 demo: Applying best k8s dev practices
+# Applying best Kubernetes development practices
+This repo is used in the Google Cloud Next '21 demo: Applying best Kubernetes development practices.
```
[the remark this answers is not in the document]
Comment (kelsk) [reply README.md]: Aha of course, thanks! Done

File: README.md (hunk 1 above)
Comment (kelsk) [reply README.md]: Done!

**Pull request merged**